subprojects/core/src/main/java/org/gradle/api/internal/plugins/ImperativeOnlyPluginApplicator.java

```diff
@@ -44,4 +44,8 @@ public void applyImperativeRulesHybrid(@Nullable String pluginId, Plugin<?> plug
         applyRules(pluginId, plugin.getClass());
     }
 
+    @Override
+    public String toString() {
+        return target.toString();
+    }
```

Comment:
The plugin manager does not have a direct reference to the target, which we want to use in the build operation name. I added these `toString()` methods to make the `pluginApplicator` usable as human-readable target string instead.